dnsserver.py

```diff
@@ -160,16 +160,16 @@ def resolve(self, request, handler):
             labelstr = str(request.q.qname)
             logger.info("requestx: %s, %s", labelstr, confs.ONLY_PRIVATE_IPS)
 
-            subdomains = labelstr.split('.')
-            if len(subdomains) == 4: # TODO: dynamic
+            if labelstr[:-1].endswith(confs.BASE_DOMAIN):
                 ip = None
+                subdomain = labelstr[:labelstr.find(confs.BASE_DOMAIN) - 1]
```

Comment:
What if `labelstr` doesn't contain `confs.BASE_DOMAIN`?

Reply:
Oh...erm...this is from  @ovidiuvio's PR, so I just  blindly submitted their code in my PR.  Do you think we should be more defensive here then?

Reply:
I think it's a good idea. otherwise we may end with `labelstr[:-2]`.

Reply:
nah, sorry, I missed the line above. It's okay.